homeassistant/components/philips_js/light.py

```diff
@@ -0,0 +1,174 @@
+        color_g /= color_c
+        color_b /= color_c
+        return color_r, color_g, color_b
+    return 0.0, 0.0, 0.0
+
+
+class PhilipsTVLightEntity(CoordinatorEntity, LightEntity):
+    """Representation of a Philips TV exposing the JointSpace API."""
+
+    def __init__(self, coordinator: PhilipsTVDataUpdateCoordinator):
+        """Initialize light."""
+        self._tv = coordinator.api
+        self._hsv = None
+        self._system = coordinator.system
+        self._coordinator = coordinator
+        super().__init__(coordinator)
+
+        self._update_from_coordinator()
+
+    @property
+    def name(self):
+        """Return the device name."""
+        return self._system["name"]
+
+    @property
+    def unique_id(self):
+        """Return unique identifier if known."""
+        return self._system["serialnumber"]
+
+    @property
+    def supported_features(self):
+        """Return supported features on this light."""
+        return SUPPORT_EFFECT | SUPPORT_COLOR | SUPPORT_BRIGHTNESS
+
+    @property
+    def should_poll(self):
+        """Device should be polled."""
+        return True
+
+    @property
+    def effect_list(self):
+        """Return the list of supported effects."""
+        return ["internal", "manual", "expert"]
+
+    @property
+    def effect(self):
+        """Return the current effect."""
+        return self._coordinator.ambilight_mode
+
+    @property
+    def hs_color(self):
+        """Return the hue and saturation color value [float, float]."""
+        if self._hsv:
+            return self._hsv[:2]
+
+    @property
+    def brightness(self):
+        """Return the hue and saturation color value [float, float]."""
+        if self._hsv:
+            return self._hsv[2] * 255 / 100
+
+    @property
+    def is_on(self):
+        """Return if the light is turned on."""
+        return self._tv.on
+
+    @property
+    def device_info(self):
+        """Return a device description for device registry."""
+        return {
+            "name": self._system["name"],
+            "identifiers": {
+                (DOMAIN, self._system["serialnumber"]),
+            },
+            "model": self._system["model"],
+            "manufacturer": "Philips",
+            "sw_version": self._system["softwareversion"],
+        }
+
+    def _update_from_coordinator(self):
+        if self._coordinator.ambilight_mode == "internal":
+            data = self._coordinator.ambilight_processed
+        else:
+            data = self._coordinator.ambilight_cached
+
+        if data:
+            color_r, color_g, color_b = _average_pixels(data)
+            self._hsv = color_RGB_to_hsv(color_r, color_g, color_b)
+        else:
+            self._hsv = None
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._update_from_coordinator()
+        super()._handle_coordinator_update()
+
+    def turn_on(self, **kwargs) -> None:
```

Comment:
Should there be an implementation of turn_off? Perhaps it could set the color to black?